packages/tldraw/src/lib/ui/components/TldrawUiLayout/TldrawUiLayout.tsx

```diff
@@ -0,0 +1,228 @@
+import { clamp } from '@tldraw/editor'
+import classNames from 'classnames'
+import { useCallback, useLayoutEffect, useRef } from 'react'
+import { PORTRAIT_BREAKPOINT } from '../../constants'
+import { useBreakpoint } from '../../context/breakpoints'
+
+/** @public */
+export type TldrawUiLayoutHorizontalPosition = 'left' | 'center' | 'right'
+
+/** @public */
+export type TldrawUiLayoutVerticalPosition = 'top' | 'center' | 'bottom'
+
+/** @public */
+export interface TldrawUiLayoutProps {
+	topLeft?: React.ReactNode
+	topCenter?: React.ReactNode
+	topRight?: React.ReactNode
+	centerLeft?: React.ReactNode
+	centerRight?: React.ReactNode
+	bottomLeft?: React.ReactNode
+	bottomCenter?: React.ReactNode
+	bottomRight?: React.ReactNode
+	spacingPx?: number
+	squishTop?: TldrawUiLayoutHorizontalPosition
```

Comment:
we should come up with a better term than squish :P
as i play around with it, it's non-intuitive as to how what will happen heh.
is there some terminology we can borrow from the `flex` or `grid` layout? or maybe something like `prefer<X>`?